app/models/host.rb

```diff
@@ -1330,69 +1330,7 @@ def scan_from_queue(taskid = nil)
-          return
-        end
-
-        update_ssh_auth_status! if respond_to?(:update_ssh_auth_status!)
-
-        if missing_credentials?
-          _log.warn("No credentials defined for #{log_target}")
-          task.update_status("Finished", "Warn", "Scanning incomplete due to Credential Issue")  if task
-          return
-        end
-
-        begin
-          connect_ssh do |ssu|
-            _log.info("Refreshing Patches for #{log_target}")
-            task.update_status("Active", "Ok", "Refreshing Patches") if task
-            Benchmark.realtime_block(:refresh_patches) { refresh_patches(ssu) }
-
-            _log.info("Refreshing Services for #{log_target}")
-            task.update_status("Active", "Ok", "Refreshing Services") if task
-            Benchmark.realtime_block(:refresh_services) { refresh_services(ssu) }
-
-            _log.info("Refreshing Linux Packages for #{log_target}")
-            task.update_status("Active", "Ok", "Refreshing Linux Packages") if task
-            Benchmark.realtime_block(:refresh_linux_packages) { refresh_linux_packages(ssu) }
-
-            _log.info("Refreshing User Groups for #{log_target}")
-            task.update_status("Active", "Ok", "Refreshing User Groups") if task
-            Benchmark.realtime_block(:refresh_user_groups) { refresh_user_groups(ssu) }
-
-            _log.info("Refreshing SSH Config for #{log_target}")
-            task.update_status("Active", "Ok", "Refreshing SSH Config") if task
-            Benchmark.realtime_block(:refresh_ssh_config) { refresh_ssh_config(ssu) }
-
-            _log.info("Refreshing FS Files for #{log_target}")
-            task.update_status("Active", "Ok", "Refreshing FS Files") if task
-            Benchmark.realtime_block(:refresh_fs_files) { refresh_fs_files(ssu) }
-
-            if supports?(:refresh_network_interfaces)
-              _log.info("Refreshing network interfaces for #{log_target}")
-              task.update_status("Active", "Ok", "Refreshing network interfaces") if task
-              Benchmark.realtime_block(:refresh_network_interfaces) { refresh_network_interfaces(ssu) }
-            end
-
-            # refresh_openstack_services should run after refresh_services and refresh_fs_files
-            if respond_to?(:refresh_openstack_services)
-              _log.info("Refreshing OpenStack Services for #{log_target}")
-              task.update_status("Active", "Ok", "Refreshing OpenStack Services") if task
-              Benchmark.realtime_block(:refresh_openstack_services) { refresh_openstack_services(ssu) }
-            end
-
-            save
-          end
-        rescue Net::SSH::HostKeyMismatch
-          # Keep from dumping stack trace for this error which is sufficiently logged in the connect_ssh method
-        rescue => err
-          _log.log_backtrace(err)
-        end
-      end
+      scan_via_ssh(task) unless is_vmware_esxi?
```

Comment:
should the check for vmware_esxi be a guard inside the `scan_via_ssh` method?

Reply:
I wonder if we could drop the `is_vmware_esxi` check and just check to see if we have a valid ssh_keypair authentication record, @Fryguy ?

Reply:
yeah i like that better - it's more important that we have creds rather than what the specific system type is.

Reply:
how does one check for valid ssh_keypair authentication record?  Is there a code snippet you can point me at?

Reply:
`authentication_status_ok?` is the standard way to check